test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -437,7 +518,19 @@ object Assertion extends AssertionVariants {
     }
 
   /**
-   * Makes a new assertion that requires the size of an iterable be satisfied
+   * Makes a new assertion that requires an Iterable to have all the elements
+   * as the specified Iterable, though not necessarily in the same order
+   */
+  def hasAllElements[A](other: Iterable[A]): Assertion[Iterable[A]] =
```

Comment:
With this assertion, the other iterable should be a subset of the tested iterable.

Reply:
I see. Maybe `hasSubset`? I think this could be implemented in terms of the intersection operator as well, basically asserting that the intersection of the actual and `other` is equal to `other`.

Reply:
I have now added `hasSubset` using `hasIntersection`.